src/AmqpBundle/Amqp/Producer.php

```diff
@@ -50,14 +50,15 @@ public function publishMessage($message, $flags = AMQP_NOPARAM, array $attribute
                       (empty($this->exchangeOptions['publish_attributes']) ? $attributes :
                       array_merge($this->exchangeOptions['publish_attributes'], $attributes));
 
-        if (empty($routingKeys)) {
-            $routingKeys = $this->exchangeOptions['routing_keys'];
+        $routingKeys = !empty($routingKeys) ? $routingKeys : $this->exchangeOptions['routing_keys'];
+        if (!$routingKeys) {
+            return $this->call($this->exchange, 'publish', [$message, null, $flags, $attributes]);
         }
 
         // Publish the message for each routing keys
-        $success = true;
+        $success = false;
         foreach ($routingKeys as $routingKey) {
-            $success &= $this->call($this->exchange, 'publish', [$message, $routingKey, $flags, $attributes]);
+            $success = $this->call($this->exchange, 'publish', [$message, $routingKey, $flags, $attributes]);
```

Comment:
It's not the same. With the "&", $success will be true only if all call return true.

```bash
$ php -a
Interactive shell

php > $success = true;
php > $success &= true;
php > echo $success;
1
php > $success &= false;
php > echo $success;
0
php > $success &= true;
php > echo $success;
0
```